### DISCUSSION TITLE
Is there a way to render on the fly? (large number of pages >1 million)

### DISCUSSION THREAD

**Discussion by Aditya94A** [gatsbyjs/gatsby#27678] [Help] [answered]
My website has a large number of pages (>1 million), so pre-rendering all at once in usual gatsby fashion would take too much time and space.

I'm thinking, is there a way to do this on the fly? When a request comes in for a page, I'd like to render it then and there and return the newly rendered page.

**Comment by herecydev** [comment, marked as answer]
No, gatsby has no server and therefore can't process requests that way. That's such a high number of pages that I would suggest looking into gatsby cloud (incremental builds). The idea being that only pages whose content has changed would need rebuilding, for example a CMS change might only require a handful of pages to update.

Otherwise youre best looking into a more traditional SSR framework like next.js
